processing/src/main/java/org/apache/druid/query/topn/NumericTopNMetricSpec.java

```diff
@@ -135,12 +133,7 @@ public TopNResultBuilder getResultBuilder(
   @Override
   public byte[] getCacheKey()
   {
-    byte[] metricBytes = StringUtils.toUtf8(metric);
-
-    return ByteBuffer.allocate(1 + metricBytes.length)
-                     .put(CACHE_TYPE_ID)
-                     .put(metricBytes)
-                     .array();
+    return new byte[] {CACHE_TYPE_ID};
```

Comment:
I removed this because it seemed redundant to have the name here after modifying the topN query toolchest to be appending the information of the ordering column immediately after, but I think your suggestion to push the query into the cache key method makes sense so that it could lookup the information itself to build the limit spec correctly would maybe make it more obvious.